edge/cmd/edgecore/app/server.go

```diff
@@ -90,15 +91,30 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 			// Get edge node local ip only when the custiomInterfaceName has been set.
 			// Defaults to the local IP from the default interface by the default config
 			if config.Modules.Edged.CustomInterfaceName != "" {
-				klog.Infof("Getting IP address by custom interface: %s", config.Modules.Edged.CustomInterfaceName)
 				ip, err := netutil.ChooseBindAddressForInterface(config.Modules.Edged.CustomInterfaceName)
 				if err != nil {
-					klog.Errorf("Failed to get IP address by custom interface: %s: %s", config.Modules.Edged.CustomInterfaceName, err)
+					klog.Errorf("Failed to get IP address by custom interface %s, err: %v", config.Modules.Edged.CustomInterfaceName, err)
 					os.Exit(1)
 				}
 				config.Modules.Edged.NodeIP = ip.String()
+				klog.Infof("Get IP address by custom interface successfully, %s: %s", config.Modules.Edged.CustomInterfaceName, config.Modules.Edged.NodeIP)
+			} else {
+				ip := net.ParseIP(config.Modules.Edged.NodeIP)
+				if ip != nil {
```

Comment:
```suggestion
				if net.ParseIP(config.Modules.Edged.NodeIP) != nil {
```

We can use this directly.

Reply:
Done